docs/modules/environment.md

```diff
@@ -84,6 +84,226 @@ function #bs.environment:get_temperature {scale:1000}
 
 ---
 
+### Celestial Movement
+
+:::::{tab-set}
+::::{tab-item} Get Current Sun Angle
+
+```{function} #bs.environment:get_current_sun_angle
+
+Get the current sun's angle on the Y axis relative to the horizon, in degrees.
+
+:Inputs:
+  **Function macro**:
+  :::{treeview}
+  - {nbt}`compound` Arguments
+    - {nbt}`number` **scale**: A scalar applied to the output.
+  :::
+
+:Outputs:
+  **Return**: Sun's angle in degrees (scaled by `scale`).
+
+  **Storage `bs:out environment.celestial_angle`** {nbt}`double`: Sun's angle in degrees.
+```
+
+*Example: Get the sun's current angle:*
+
+```mcfunction
+# Once
+function #bs.environment:get_sun_angle {scale: 1000}
+tellraw @a [{"text":"Sun angle: "},{"nbt":"environment.celestial_angle","storage":"bs:out","interpret":true},{"text":"°"}]
+```
+
+```{admonition} doDaylightCycle gamerule
+:class: info
+
+This feature also works when the doDaylightCycle gamerule is set to false.
+In such situation, the returned value is the sun's angle at the time of daylight cycle freeze.
+```
+
+::::
+::::{tab-item} Get Current Moon Angle
+
+```{function} #bs.environment:get_current_moon_angle
+
+Get the current moon's angle on the Y axis relative to the horizon, in degrees.
+
+:Inputs:
+  **Function macro**:
+  :::{treeview}
+  - {nbt}`compound` Arguments
+    - {nbt}`number` **scale**: A scalar applied to the output.
+  :::
+
+:Outputs:
+  **Return**: Moon's angle in degrees (scaled by `scale`).
+
+  **Storage `bs:out environment.celestial_angle`** {nbt}`double`: Moon's angle in degrees.
+```
+
+*Example: Get the moon's current angle:*
+
+```mcfunction
+# Once
+function #bs.environment:get_moon_angle {scale: 1000}
+tellraw @a [{"text":"Moon angle: "},{"nbt":"environment.celestial_angle","storage":"bs:out","interpret":true},{"text":"°"}]
+```
+
+```{admonition} doDaylightCycle gamerule
+:class: info
+
+This feature also works when the doDaylightCycle gamerule is set to false.
+In such situation, the returned value is the moon's angle at the time of daylight cycle freeze.
+```
+
+::::
+::::{tab-item} Get Sun Angle
+
+```{function} #bs.environment:get_sun_angle
+
+Get the sun's angle on the Y axis relative to the horizon at a specific time.
+
+:Inputs:
+  **Function macro**:
+  :::{treeview}
+  - {nbt}`compound` Arguments
+    - {nbt}`int` **day**: The day number.
+    - {nbt}`int` **daytime**: The time of day in ticks (0-24000).
```

Comment:
Im really not convinced by those params. It makes the function hard to use. Since those are ints i think we should use input scores for both daytime and time. Sure im assuming that scale is mostly static and that daytime and time are computed and change a lot but i think this is a fair assumption. Also even if we mix input types i think it remains a simple signature and avoid passing around a macro. This is a bit like in get and set pos in the position module where we use scores with the scale param. Last argument is that with the current impl it will encourage datapackers to use the internal function without calling the function tag (which is something i've already seen people doing and that i really think we should avoid)